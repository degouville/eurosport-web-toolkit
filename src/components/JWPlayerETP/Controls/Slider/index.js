import React from 'react';
import RCSlider from 'rc-slider';
import PropTypes from 'prop-types';

const getHandleStyle = ({ handleSize, railThickness, handleColor }) => ({
  borderColor: 'transparent',
  height: handleSize,
  width: handleSize,
  backgroundColor: handleColor,
  margin: -(handleSize / 2) + railThickness / 2,
});

const getBarStyle = ({ backgroundColor, railThickness, vertical }) => {
  if (vertical) return { backgroundColor, width: railThickness };
  return { backgroundColor, height: railThickness };
};

const Slider = ({
  value,
  onChange,
  defaultValue,
  min,
  max,
  vertical,
  railThickness,
  railColor,
  handleSize,
  handleColor,
  trackColor,
}) => (
  <RCSlider
    defaultValue={defaultValue}
    onChange={onChange}
    vertical={vertical}
    trackStyle={getBarStyle({ backgroundColor: trackColor, railThickness, vertical })}
    handleStyle={getHandleStyle({ handleSize, railThickness, handleColor })}
    railStyle={getBarStyle({ backgroundColor: railColor, railThickness, vertical })}
    value={value}
    min={min}
    max={max}
  />
);

Slider.propTypes = {
  // Values
  // eslint-disable-next-line react/forbid-prop-types
  min: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  defaultValue: PropTypes.number.isRequired,

  // Theme
  vertical: PropTypes.bool.isRequired,
  railThickness: PropTypes.number.isRequired,
  railColor: PropTypes.string.isRequired,
  handleSize: PropTypes.number.isRequired,
  handleColor: PropTypes.string.isRequired,
  trackColor: PropTypes.string.isRequired,
};

export default Slider;
