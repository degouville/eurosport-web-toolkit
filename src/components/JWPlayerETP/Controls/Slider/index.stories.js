import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';
import theme from 'src/theme';
import Slider from './index';

const stories = storiesOf('Components|JWPlayerETP', module);

const SliderControl = props => {
  const [progress, setProgress] = useState(50);
  return <Slider value={progress} onChange={setProgress} defaultValue={50} {...props} />;
};

stories.add('Slider horizontal', () => <SliderControl {...theme.playerControls.seek} />);
stories.add('Slider vertical', () => (
  <div style={{ width: 500, height: 500 }}>
    <SliderControl {...theme.playerControls.volume} />
  </div>
));
