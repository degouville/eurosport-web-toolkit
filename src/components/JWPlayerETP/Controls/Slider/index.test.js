import React from 'react';
import { shallow } from 'enzyme';
import Slider from './index';

describe('Components|JWPlayerETP|Slider', () => {
  const createProps = newProps => ({
    value: 50,
    onChange: jest.fn(),
    defaultValue: 0,
    min: 0,
    max: 50,
    railThickness: 10,
    railColor: 'blue',
    handleSize: 20,
    handleColor: 'red',
    trackColor: 'green',
    ...newProps,
  });

  it('should match snapshot on vertical', () => {
    const props = createProps({ vertical: true });
    const wrapper = shallow(<Slider {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot on horizontal', () => {
    const props = createProps({ vertical: false });
    const wrapper = shallow(<Slider {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
