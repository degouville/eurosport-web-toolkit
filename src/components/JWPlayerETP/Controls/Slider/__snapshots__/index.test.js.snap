// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components|JWPlayerETP|Slider should match snapshot on horizontal 1`] = `
<ComponentEnhancer(undefined)
  activeDotStyle={Object {}}
  className=""
  defaultValue={0}
  disabled={false}
  dotStyle={Object {}}
  dots={false}
  handle={[Function]}
  handleStyle={
    Object {
      "backgroundColor": "red",
      "borderColor": "transparent",
      "height": 20,
      "margin": -5,
      "width": 20,
    }
  }
  included={true}
  marks={Object {}}
  max={50}
  min={0}
  onAfterChange={[Function]}
  onBeforeChange={[Function]}
  onChange={[MockFunction]}
  prefixCls="rc-slider"
  railStyle={
    Object {
      "backgroundColor": "blue",
      "height": 10,
    }
  }
  step={1}
  trackStyle={
    Object {
      "backgroundColor": "green",
      "height": 10,
    }
  }
  value={50}
  vertical={false}
/>
`;

exports[`Components|JWPlayerETP|Slider should match snapshot on vertical 1`] = `
<ComponentEnhancer(undefined)
  activeDotStyle={Object {}}
  className=""
  defaultValue={0}
  disabled={false}
  dotStyle={Object {}}
  dots={false}
  handle={[Function]}
  handleStyle={
    Object {
      "backgroundColor": "red",
      "borderColor": "transparent",
      "height": 20,
      "margin": -5,
      "width": 20,
    }
  }
  included={true}
  marks={Object {}}
  max={50}
  min={0}
  onAfterChange={[Function]}
  onBeforeChange={[Function]}
  onChange={[MockFunction]}
  prefixCls="rc-slider"
  railStyle={
    Object {
      "backgroundColor": "blue",
      "width": 10,
    }
  }
  step={1}
  trackStyle={
    Object {
      "backgroundColor": "green",
      "width": 10,
    }
  }
  value={50}
  vertical={true}
/>
`;
